export_dagmc_cmd/DAGMCExportCommand.cpp

```diff
@@ -0,0 +1,856 @@
+  CHK_MB_ERR_RET_MB("Error creating id_tag",rval);
+  
+  rval = mdbImpl->tag_get_handle(NAME_TAG_NAME, NAME_TAG_SIZE, moab::MB_TYPE_OPAQUE,
+                                 name_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating name_tag",rval);
+  
+  rval = mdbImpl->tag_get_handle(CATEGORY_TAG_NAME, CATEGORY_TAG_SIZE, moab::MB_TYPE_OPAQUE,
+                                 category_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating category_tag",rval);
+
+  rval = mdbImpl->tag_get_handle("FACETING_TOL", 1, moab::MB_TYPE_DOUBLE, faceting_tol_tag,
+                                 moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating faceting_tol_tag",rval);
+
+  rval = mdbImpl->tag_get_handle("GEOMETRY_RESABS", 1, moab::MB_TYPE_DOUBLE, 
+                                 geometry_resabs_tag, moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+  CHK_MB_ERR_RET_MB("Error creating geometry_resabs_tag",rval);
+
+  return rval;
+}
+
+void DAGMCExportCommand::teardown()
+{
+  message  << "***** Faceting Summary Information *****" << std::endl;
+  if (0 < failed_curve_count) {
+    message << "----- Curve Fail Information -----" << std::endl
+            << "There were " << failed_curve_count << " curves that could not be faceted." << std::endl;
+  } else {
+    message << "----- All curves faceted correctly  -----" << std::endl;
+  }
+  if (0 < failed_surface_count) {
+    message << "----- Facet Fail Information -----" << std::endl
+            << "There were " << failed_surface_count << " surfaces that could not be faceted." << std::endl;
+  } else {
+    message << "----- All surfaces faceted correctly  -----" << std::endl;
+  }
+  message << "***** End of Faceting Summary Information *****" << std::endl;
+ 
+  CubitInterface::get_cubit_message_handler()->print_message(message.str().c_str()); 
+  message.str("");
+  delete myGeomTool;
+  delete mdbImpl;
+
+}
+
+
+moab::ErrorCode DAGMCExportCommand::create_entity_sets(refentity_handle_map (&entmap)[5])
+{
+
+  moab::ErrorCode rval;
+
+  const char geom_categories[][CATEGORY_TAG_SIZE] =
+              {"Vertex\0", "Curve\0", "Surface\0", "Volume\0", "Group\0"};
+  const char* const names[] = {"Vertex", "Curve", "Surface", "Volume"};
+  DLIList<RefEntity*> entlist;
+
+  for (int dim = 0; dim < 4; dim++) {
+    entlist.clean_out();
+    GeometryQueryTool::instance()->ref_entity_list(names[dim], entlist, true);
+    entlist.reset();
+
+    message << "Found " << entlist.size() << " entities of dimension " << dim << std::endl;
+
+    for (int i = entlist.size(); i--;) {
+      RefEntity* ent = entlist.get_and_step();
+      moab::EntityHandle handle;
+
+      // Create the new meshset
+      rval = mdbImpl->create_meshset(dim == 1 ? moab::MESHSET_ORDERED : moab::MESHSET_SET, handle);
+      if (moab::MB_SUCCESS != rval) return rval;
```

Comment:
Using the same error macros throughout would be nice so that errors can be seen in the Trelis command window. They've already helped us track down a couple of bugs.